thrust/thrust/system/tbb/detail/scan.inl

```diff
@@ -47,6 +47,100 @@ namespace detail
 namespace scan_detail
 {
 
+template <typename InputIterator, typename OutputIterator, typename BinaryFunction, typename ValueType>
+struct inclusive_body_init
```

Comment:
I am not convinced that using a standalone body just for `init` is a redundancy. There should be a way for `inclusive_body_init` and `inclusive_body` to be merged.

Reply:
I pushed the change to your branch (and this PR).

Reply:
@gonidelis I pushed a unification of `inclusive_body_init` with the below `inclusive_body` to your branch. It seems you force pushed over it and deleted my changes :( Can you please unify those two structs again? IIRC, it takes a single `_CCCL_IF_CONSTEXPR` in one place.